--- cronie-1.4.1/src/cron-paths.h~	2009-08-11 22:01:57.847314880 +0200
+++ cronie-1.4.1/src/cron-paths.h	2009-08-11 22:06:36.364162561 +0200
@@ -17,11 +17,11 @@
 		/* cron allow/deny file.  At least cron.deny must
 		 * exist for ordinary users to run crontab.
 		 */
-#define	CRON_ALLOW	"/etc/cron.allow"
-#define	CRON_DENY	"/etc/cron.deny"
+#define	CRON_ALLOW	"/etc/cron/cron.allow"
+#define	CRON_DENY	"/etc/cron/cron.deny"
 
 		/* 4.3BSD-style crontab f.e. /etc/crontab */
-#define SYSCRONTAB	"/etc/crontab"
+#define SYSCRONTAB	"/etc/cron.d/crontab"
 
 		/* system crontab dir f.e. /etc/cron.d/ */
 #define SYS_CROND_DIR	"/etc/cron.d"
